flink-runtime/src/main/java/org/apache/flink/runtime/jobmaster/JobManagerRunner.java

```diff
@@ -96,6 +97,15 @@ public interface JobManagerRunner extends AutoCloseableAsync {
      */
     CompletableFuture<ExecutionGraphInfo> requestJob(Time timeout);
 
+    /**
+     * Requests the {@link CheckpointStatsSnapshot} of the executed job.
+     *
+     * @param timeout for the rpc call
+     * @return Future which is completed with the {@link CheckpointStatsSnapshot} of the executed
+     *     job. Returns null if job is not running.
+     */
+    CompletableFuture<CheckpointStatsSnapshot> requestCheckpointStats(Time timeout);
```

Comment:
Why is this needed?

Reply:
This particular change is required if we want to just request the `CheckpointStatsSnapshot` object from the JobMaster to side step the recursive copy of the ArchivedExecutionGraph on the JM side.

Currently, the process is:
Client -> REST (cache ExecutionGraph) -> JobManager -> JobMaster (builds ArchivedExecutionGraph - recursive copy, costly(?))

Without this change, the process will be
Client -> REST (cache CheckpointStatsSnapshot) -> JobManager (request ExecutionGraphInfo) -> JobMaster (builds ArchivedExecutionGraph - recursive copy, costly (?))

With this change, the process will be
Client -> REST (cache CheckpointStatsSnapshot) -> JobManager (request CheckpointStatsSnapshot) -> JobMaster (retrieves CheckpointStatsSnapshot from cache in CheckpointStatsTracker)


I think this change can be helpful to refactor the `JobManager` / `JobMaster` API to retrieve specific information about a job, rather than 1 large data dump (ExecutionGraph)! Let me know what you think

Reply:
The point is that you don't need to introduce a new method in JobManagerRunner, but instead, Dispatcher can directly talk to the JobMasterGateway (`performOperationOnJobMasterGateway` is what you should look for).

Reply:
Ah. Got it! Will refactor